src/python/pants/backend/java/tailor.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.backend.java.target_types import JavaLibrary, JavaTestsSources, JunitTests
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsRequest,
+    group_by_dir,
+)
+from pants.engine.fs import PathGlobs, Paths
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+from pants.source.filespec import Filespec, matches_filespec
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class PutativeJavaTargetsRequest(PutativeTargetsRequest):
```

Comment:
I had to go look up the word `Putative`, but if that's what we use elsewhere then I guess it's an OK name

Reply:
That's what @benjyw chose when he wrote `tailor`. Note that `PutativeTargetsRequest` is part of the core rules, so this matching this naming scheme is the proper course.

Reply:
Yeah, that's fine